File: networking/src/main/java/com/jbrunton/networking/resources/movies/MovieSearchResultResource.java
package com.jbrunton.networking.resources.movies;

import com.google.common.base.Optional;
import com.jbrunton.entities.Configuration;
import com.jbrunton.entities.Movie;

public class MovieSearchResultResource extends BaseMovieResource {
    public Movie toMovie(Configuration config) {
        return builder(config)
                .overview(Optional.absent())
                .build();
    }
}

File: networking/src/main/java/com/jbrunton/mymovies/networking/resources/movies/MovieSearchResultResource.kt
package com.jbrunton.mymovies.networking.resources.movies

import com.jbrunton.mymovies.entities.models.Configuration
import com.jbrunton.mymovies.entities.models.Movie
import org.joda.time.LocalDate

class MovieSearchResultResource(
        override val id: String,
        override val originalTitle: String,
        override val posterPath: String,
        override val backdropPath: String,
        override val releaseDate: LocalDate?,
        override val voteAverage: String) : BaseMovieResource {

    fun toMovie(config: Configuration): Movie {
        return Movie(
                id = id,
                title = originalTitle,
                posterUrl = config.expandUrl(posterPath),
                backdropUrl = config.expandUrl(backdropPath),
                releaseDate = releaseDate,
                rating = voteAverage,
                overview = null
        )
    }
}
